src/pool/srv_pool.c

```diff
@@ -1148,7 +1148,7 @@ events_handler(void *arg)
 	D_DEBUG(DB_MD, DF_UUID": starting\n", DP_UUID(svc->ps_uuid));
 
 	for (;;) {
-		struct pool_svc_event  *event;
+		struct pool_svc_event  *event = NULL;
```

Comment:
seems like a reasonable change. Was there a situation in which a stale / already freed event was processed?

Reply:
I have an older compiler that I was using and was giving me `error: ‘event’ may be used uninitialized in this function [-Werror=maybe-uninitialized]`. I don't have to commit this change but thought it was a good one. If the event pointer is never set within the if statement, then the handle_event would receive an uninitialized pointer.

Reply:
I don't think it can actually happen, but if 'stop' == false and d_list_empty(...) is true??

Reply:
In that case we will call sched_cond_wait and check the "stop" and "d_list_empty" condition again, that is, we will not break the inner loop.